### PR TITLE
(BKR-1484) Packaging platform overrides for puppet5 install utils

### PR DESCRIPTION
**Pull request by caseywilliams**
Allows the environment variable BEAKER_PACKAGING_PLATFORMS to override
the default packaging platforms specified for hosts by
beaker-hostgenerator in `host_urls`. BEAKER_PACKAGING_PLATFORMS should
be a comma-separated list with entries of the format
`<platform-name>=<packaging-platform-name>`.

[Here's an example run of puppet-agent and puppet acceptance](https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/ad-hoc/job/platform_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/204/), with `
BEAKER_PACKAGING_PLATFORMS="aix-7.1-power=aix-6.1-power,aix-7.2-power=aix-6.1-power"` set on puppet-agent only; The puppet-agent tests install the 6.1 package on 7.1 and 7.2, and the puppet tests (which don't use the environment variable) fail, because they still look for 7.1 and 7.2 packages, which aren't available in this run.

### CODE REVIEW

File: lib/beaker-puppet/install_utils/puppet5.rb [hunk 1]
```diff
@@ -27,7 +27,29 @@ def fetch_build_details(sha_yaml_url)
...
+      # entries of the format `<host-platform>=<override-platform>`
+      #
+      # @param [Host] host Host whose packaging platform to determine
+      # @return [String] The packaging platform
+      def host_packaging_platform(host)
+        packaging_platform = host[:packaging_platform]
+        if ENV['BEAKER_PACKAGING_PLATFORMS']
+          overrides = Hash[ENV['BEAKER_PACKAGING_PLATFORMS'].split(',').map { |e| e.split('=') }]
+          logger.debug("Found packaging platform overrides: #{overrides}")
+          if overrides[host[:packaging_platform]]
```
Comment (branan) [inline lib/beaker-puppet/install_utils/puppet5.rb]: Shouldn't this be looking up overrides for `host[:platform]`?

Comment (caseywilliams) [reply lib/beaker-puppet/install_utils/puppet5.rb]: Yeah, it should -- my aix tests just happened to work out for that specific case anyway. I'll update.

**Comment by caseywilliams** [comment]
The PR test failure looks transient

**Comment by caseywilliams** [comment]
jenkins please test this

**Pull request merged**